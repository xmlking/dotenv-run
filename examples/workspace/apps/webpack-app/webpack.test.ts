import { execSync } from "child_process";
import { describe, it, expect } from "vitest";

describe("dotenv-run should give the expected output", () => {
  it("empty", () => {
    const actual = execSync(
      "node dist/main.js",
      { encoding: "utf8" }
    );
    expect(actual).toMatchSnapshot();
  });

  it("dev", () => {
    const actual = execSync(
      "NODE_ENV=dev node dist/main.js",
      { encoding: "utf8" }
    );
    expect(actual).toMatchSnapshot();
  });

  it("prod", () => {
    const actual = execSync(
      "NODE_ENV=prod node dist/main.js",
      { encoding: "utf8" }
    );
    expect(actual).toMatchSnapshot();
  });
});
