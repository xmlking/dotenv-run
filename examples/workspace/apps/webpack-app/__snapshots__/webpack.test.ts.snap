// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`dotenv-run should give the expected output > dev 1`] = `
"https://dotenv-run.app/api/v1/users
production
"
`;

exports[`dotenv-run should give the expected output > empty 1`] = `
"https://dotenv-run.app/api/v1/users
production
"
`;

exports[`dotenv-run should give the expected output > prod 1`] = `
"https://dotenv-run.app/api/v1/users
production
"
`;
